script.js

```diff
@@ -1 +1,260 @@
+playAgainButton.classList.add('button');
+playAgainButton.innerText = 'Play Again';
+
+// global for selectors container
+const selectorCont = document.createElement('div');
+selectorCont.classList.add('message');
+
+const boardSizeSelector = document.createElement('div');
+boardSizeSelector.classList.add('selector');
+boardSizeSelector.innerHTML = 'Board Size<br>';
+
+// global for input field: board size
+const boardSizeInput = document.createElement('input');
+boardSizeInput.setAttribute('type', 'number');
+boardSizeInput.value = 3;
+boardSizeInput.classList.add('input');
+
+console.log(typeof boardSizeInput.value);
+
+// global for input + button
+const boardSizePlusButton = document.createElement('button');
+boardSizePlusButton.innerText = '+';
+
+// global for input - button
+const boardSizeMinusButton = document.createElement('button');
+boardSizeMinusButton.innerText = '-';
+
+// let squareElement;
+
+// ============== HELPER FUNCTIONS =================
+// completely rebuilds the entire board every time there's a click
+const buildBoard = (board) => {
+  // start with an empty container
+  boardContainer.innerHTML = '';
+  boardElement = document.createElement('div');
+  boardElement.classList.add('board');
+
+  // move through the board data array and create the
+  // current state of the board
+  for (let i = 0; i < board.length; i += 1) {
+    // separate var for one row / row element
+    const row = board[i];
+    const rowElement = document.createElement('div');
+    rowElement.classList.add('row');
+
+    // set each square
+    // j is the column number
+    for (let j = 0; j < row.length; j += 1) {
+      // one square element
+      square = document.createElement('div');
+      square.classList.add('square');
+
+      // set the text of the square according to the array
+      square.innerText = board[i][j];
+
+      // set the click all over again
+      // eslint-disable-next-line
+      square.addEventListener('click', () => {
+        squareClick(i, j);
+      });
+
+      rowElement.appendChild(square);
+    }
+
+    // add a single row to the board
+    boardContainer.appendChild(rowElement);
+  }
+};
+
+// function to append text to message container
+const output = (message) => {
+  messageCont.innerHTML = message;
+};
+
+const plusButtonClick = (target) => {
+  const currentValue = Number(target.value);
+  target.value = currentValue + 1;
+  buildEmptyBoard(target.value);
+  buildBoard(board);
+};
+
+const minusButtonClick = (target) => {
+  const currentValue = Number(target.value);
+  if (currentValue >= 4) {
+    target.value = currentValue - 1; }
+  buildEmptyBoard(target.value);
+  buildBoard(board);
+};
+
+// =============== GAME INITIALISATION LOGIC =================
+// create the board container element and put it on the screen
+const initGame = () => {
+  // append message Container with defaulty message
+  output('Win by plotting your Xs or Os in a<br>diagonal, horizontal or vertical manner');
```

Comment:
consider making a variable for default message to avoid repetition:
```
const defaultMsg = 'Win by plotting your Xs or Os in a<br>diagonal, horizontal or vertical manner';
output(defaultMsg);
```